### PR TITLE
1133 twitter GitHub input validation

### LINKED ISSUE #1133: project verification: add input validation for twitter & github fields (no @, no URLs, etc.)

### PR DESCRIPTION
**Pull request by KurtMerbeth**
fixes #1133 

image

Deployment URL: https://bafybeifygyz7xbp5a2j7bkh7dc2k2ezfrx34efkyelni4hacx46maj3sbe.on.fleek.co/

### CODE REVIEW

**Review by thelostone-mc — APPROVED** [review]

**Comment by KurtMerbeth** [comment]
> Hey Kurt! This overall looks great. Just going to call out some minor bugs/changes:
> 
> * [ ]  It looks like this change made the project description box smaller (on production, the description box is larger)
> 
> Screen Shot 2023-03-16 at 10 39 45 AM
> 

the bug is on staging too. so the root cause is not my PR. But i will look into it

**Comment by KurtMerbeth** [comment]
image

File: packages/builder/src/components/base/FormValidationErrorList.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+export default function FormValidationErrorList({ formValidation }: any) {
+  if (!formValidation.valid && formValidation.errorCount > 0)
+    return (
+      <div
```
Comment (DanieleSalatti) [inline packages/builder/src/components/base/FormValidationErrorList.tsx:4]: Nit: add `aria-live="assertive"`

Although I don't think we have cared much about this kind of things so far...

File: packages/builder/src/components/base/FormValidationErrorList.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export default function FormValidationErrorList({ formValidation }: any) {
```
Comment (DanieleSalatti) [inline packages/builder/src/components/base/FormValidationErrorList.tsx:1]: Nit: consider replacing `any` with an interface:

```
interface FormValidation {
  valid: boolean;
  errorCount: number;
  messages: string[];
}
```

(untested)

File: packages/builder/src/components/base/FormValidationErrorList.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      <div
+        className="p-4 text-gitcoin-pink-500 border rounded border-red-900/10 bg-gitcoin-pink-100 mt-8"
+        role="alert"
+      >
+        <strong className="text-gitcoin-pink-500 font-medium text-sm">
+          There {formValidation.errorCount === 1 ? "was" : "were"}{" "}
+          {formValidation.errorCount}{" "}
+          {formValidation.errorCount === 1 ? "error" : "errors"} with your form
+          submission
+        </strong>
```
Comment (DanieleSalatti) [inline packages/builder/src/components/base/FormValidationErrorList.tsx:13]: Something along these lines might be a bit easier to read (untested):

```
<strong className="text-gitcoin-pink-500 font-medium text-sm">
  There {formValidation.errorCount === 1 ? "was 1 error" : `were ${formValidation.errorCount} errors`} with your form submission
</strong>
```

File: packages/builder/src/components/base/VerificationForm.tsx [hunk 4]
```diff
@@ -24,7 +27,16 @@ export default function VerificationForm({
...
+  const [failed, setFailed] = useState<string | undefined>();
+  const [formValidation, setFormValidation] = useState({
+    messages: [""],
+    valid: true,
+    errorCount: 0,
+  });
+
+  const [feedback, setFeedback] = useState([
+    { title: "", type: "none", message: "" },
+  ]);
```
Comment (DanieleSalatti) [inline packages/builder/src/components/base/VerificationForm.tsx:39]: Nit: we could use interfaces/types here to:

```
interface FormValidation {
  ...
}

const [formValidation, setFormValidation] = useState<FormValidation>({
  messages: [""],
  valid: true,
  errorCount: 0,
});
```

Comment (DanieleSalatti) [reply packages/builder/src/components/base/VerificationForm.tsx:39]: Also, consider using an enum for the `type`.

**Review by DanieleSalatti — APPROVED** [review]
Approved (comments are nits)

**Comment by willsputra** [comment]
the updated copy looks good to me!

**Review by nfrgosselin — APPROVED** [review]

**Pull request merged**